src/server/cookies.ts

```diff
@@ -250,19 +262,26 @@ export function setChunkedCookie(
  */
 export function getChunkedCookie(
   name: string,
-  reqCookies: RequestCookies
+  reqCookies: RequestCookies,
+  isLegacyCookie?: boolean
 ): string | undefined {
   // Check if regular cookie exists
   const cookie = reqCookies.get(name);
   if (cookie?.value) {
+    // If the base cookie exists, return its value (handles non-chunked case)
     return cookie.value;
   }
 
-  const chunks = getAllChunkedCookies(reqCookies, name).sort(
+  const chunks = getAllChunkedCookies(
+    reqCookies,
+    name,
+    isLegacyCookie ?? false
```

Comment:
I believe we can avoid `?? false` here

```suggestion
    isLegacyCookie
```